app/api/add/router.tsx

```diff
@@ -0,0 +1,23 @@
+import { data } from "../data"
+
+async function POST(request, {params}) {
+
+    const body = await request.json()
+
+    const new_todo = {
+        id: data.length + 1,
+        todo: body.todo,
+        author: body.author,
+        done: false
+    }
+
+    const res = await fetch("http://localhost:3000/api/", {
```

Comment:
please add a `tag` to this fetch and have an example of cache re validation as well. Maybe re-validate the cache every time a new item is added.